src/Illuminate/Database/Eloquent/Concerns/HasRelationships.php

```diff
@@ -731,7 +731,9 @@ public function getMorphClass()
             return array_search(static::class, $morphMap, true);
         }
 
-        return static::class;
+        return Relation::$useTableNamesForMorphMap
```

Comment:
While this code is being changed, maybe it would also be useful to add an optional property/method to models (eg. `morphableName`) that can be used - so each model can represent its own morphable name itself instead of being done elsewhere in a map?